### PR TITLE
Issue#2078: For LOAD DATA LOCAL, no error is reported when the file name is empty.

### PR DESCRIPTION
**Pull request by ZhuKaley**
Issue: #2078

### CODE REVIEW

File: src/sql/resolver/cmd/ob_load_data_resolver.cpp [hunk 1]
```diff
@@ -589,7 +589,7 @@ int ObLoadDataResolver::resolve_filename(ObLoadDataStmt *load_stmt, ParseNode *n
     LOG_WARN("invalid node", "child", file_name_node);
   } else {
     ObString file_name(file_name_node->str_len_, file_name_node->str_value_);
-    if (OB_UNLIKELY(file_name.empty())) {
+    if (OB_UNLIKELY(file_name.empty()) && ObLoadFileLocation::CLIENT_DISK != load_args.load_file_storage_) {
```
Comment (hnwyllmm) [inline src/sql/resolver/cmd/ob_load_data_resolver.cpp:592]: 应该是：
if (file_name.empty()) {
  if (ObLoadFileLocation::CLIENT_DISK != load_args.load_file_storage_) {
    ret = OB_FILE_NOT_EXIST;
  } else {
    // do nothing
  }
  ...
} else {
 ...
}
??

能否帮忙确认，JDBC+MySQL中，如果输入一个根本不存在的文件名，mysql也不会校验，比如 
load data local infile '123' into table t xxxx;

**Review by hnwyllmm — CHANGES_REQUESTED** [review]

File: src/sql/resolver/cmd/ob_load_data_resolver.cpp (hunk 1 above)
Comment (ZhuKaley) [reply src/sql/resolver/cmd/ob_load_data_resolver.cpp:592]: 1、我按你的修改方法重新提交一下；
2、输入一个根本不存在的文件名，MySQL没有检查，客户端在读文件时，发现文件不存在，会报文件不存在的错误。
报错信息如下：
Exception in thread "main" java.lang.RuntimeException: java.sql.SQLException: Unable to open file 'lllllll'for 'LOAD DATA LOCAL INFILE' command.Due to underlying IOException: 

** BEGIN NESTED EXCEPTION ** 

java.io.FileNotFoundException
MESSAGE: lllllll (No such file or directory)

STACKTRACE:

java.io.FileNotFoundException: lllllll (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
        at com.mysql.cj.protocol.a.NativeProtocol.sendFileToServer(NativeProtocol.java:1757)
        at com.mysql.cj.protocol.a.TextResultsetReader.read(TextResultsetReader.java:107)
        at com.mysql.cj.protocol.a.TextResultsetReader.read(TextResultsetReader.java:48)
        at com.mysql.cj.protocol.a.NativeProtocol.read(NativeProtocol.java:1590)
        at com.mysql.cj.protocol.a.NativeProtocol.readAllResults(NativeProtocol.java:1644)
        at com.mysql.cj.protocol.a.NativeProtocol.sendQueryPacket(NativeProtocol.java:953)
        at com.mysql.cj.protocol.a.NativeProtocol.sendQueryString(NativeProtocol.java:886)
        at com.mysql.cj.NativeSession.execSQL(NativeSession.java:1073)
        at com.mysql.cj.jdbc.StatementImpl.executeInternal(StatementImpl.java:724)
        at com.mysql.cj.jdbc.StatementImpl.execute(StatementImpl.java:648)
        at Main.testMySQL(Main.java:79)
        at Main.main(Main.java:36)


** END NESTED EXCEPTION **


        at Main.testMySQL(Main.java:91)
        at Main.main(Main.java:36)
Caused by: java.sql.SQLException: Unable to open file 'lllllll'for 'LOAD DATA LOCAL INFILE' command.Due to underlying IOException: 

** BEGIN NESTED EXCEPTION ** 

java.io.FileNotFoundException
MESSAGE: lllllll (No such file or directory)

STACKTRACE:

java.io.FileNotFoundException: lllllll (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
        at com.mysql.cj.protocol.a.NativeProtocol.sendFileToServer(NativeProtocol.java:1757)
        at com.mysql.cj.protocol.a.TextResultsetReader.read(TextResultsetReader.java:107)
        at com.mysql.cj.protocol.a.TextResultsetReader.read(TextResultsetReader.java:48)
        at com.mysql.cj.protocol.a.NativeProtocol.read(NativeProtocol.java:1590)
        at com.mysql.cj.protocol.a.NativeProtocol.readAllResults(NativeProtocol.java:1644)
        at com.mysql.cj.protocol.a.NativeProtocol.sendQueryPacket(NativeProtocol.java:953)
        at com.mysql.cj.protocol.a.NativeProtocol.sendQueryString(NativeProtocol.java:886)
        at com.mysql.cj.NativeSession.execSQL(NativeSession.java:1073)
        at com.mysql.cj.jdbc.StatementImpl.executeInternal(StatementImpl.java:724)
        at com.mysql.cj.jdbc.StatementImpl.execute(StatementImpl.java:648)
        at Main.testMySQL(Main.java:79)
        at Main.main(Main.java:36)


** END NESTED EXCEPTION **


        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
        at com.mysql.cj.jdbc.StatementImpl.executeInternal(StatementImpl.java:764)
        at com.mysql.cj.jdbc.StatementImpl.execute(StatementImpl.java:648)
        at Main.testMySQL(Main.java:79)
        ... 1 more

Comment (ZhuKaley) [reply src/sql/resolver/cmd/ob_load_data_resolver.cpp]: 已重新提交

Comment (hnwyllmm) [reply src/sql/resolver/cmd/ob_load_data_resolver.cpp]: 感谢。
你的测试结果非常重要。说明在MySQL中，load local模式不会去校验文件名是否合法的，都在MySQL客户端自己做校验。为了更好的兼容性，那LOCAL模式时应该这么操作是否应该这么调整：
原来的代码：
```cpp
ObString file_name(file_name_node->str_len_, file_name_node->str_value_);
    if (OB_UNLIKELY(file_name.empty())) {
```

应该修改为：
```cpp
ObString file_name(file_name_node->str_len_, file_name_node->str_value_);
if (ObLoadFileLocation::CLIENT_DISK == load_args.load_file_storage_) {
   // 处理LOCAL模式的文件，不关心文件名是否有问题
   if (OB_FAIL(load_args.file_iter_.add_files(file_name)) {
        LOG_WARN("failed to add file to file_iter", K(ret), K(file_name));
   }
} else if (OB_UNLIKELY(file_name.empty())) {
   // 这里处理SERVER和OSS模式下的文件
   .....
```

你看看是否合理

**Review by hnwyllmm — APPROVED** [review]

**Pull request merged**